src/other/utils.jl

```diff
@@ -46,7 +46,9 @@ function makeidentifier(s::AbstractString)
         end
     end
 
-    return String(take!(res))
+    # return String(take!(res))
+    # TODO this is to fix a bug in Compat for 0.6
```

Comment:
Can you give more details? Is there an issue open in Compat.jl?

Reply:
see [this issue](https://github.com/JuliaLang/Compat.jl/issues/502).  I'm hoping it will be resolved one way or another before I'm even done with this PR...